libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -196,6 +208,31 @@ AP_Camera::trigger_pic_cleanup()
     }
 }
 
+void AP_Camera::handle_message(mavlink_channel_t chan, const mavlink_message_t &msg)
+{
+    switch (msg.msgid) {
+    case MAVLINK_MSG_ID_DIGICAM_CONTROL:
+        control_msg(msg);
+        break;
+    case MAVLINK_MSG_ID_GOPRO_HEARTBEAT:
+        // heatbeat from mavlink solo gimbal with a GoPro 3/4
+        if (_trigger_type == AP_CAMERA_TRIGGER_TYPE_GOPRO) {
+            AP_Camera_SoloGimbal::handle_gopro_heartbeat(chan, msg);
+            break;
+        }
+        break;
+    }
+}
+
+/// GoPro 3/4 in mavlink Solo Gimbal to toggle shutter
+/// pass through to AP_Camera_soloGimbal 
+void AP_Camera::gopro_capture_mode_toggle()
```

Comment:
This is probably a good candidate (in the future) to be a generic "Toggle Video/Photo Mode" option, rather than being GoPro-specific...

The fact only Solo does means we can do this down the track (@andyp1per ;-) )... but perhaps we should change the user-facing terminology now so as to save confusion later?

To be clear - *this* method would change name, the RC option would change name, the call into the `AP_Camera_SoloGimbal` library would stay the same.